feat: add crn parser submodule (#2)
terraform-ibm-modules/terraform-ibm-common-utilities: examples/crn-parser/README.md

# CRN parser example

> An example that will provision the following:
> 
> - A new resource group if one is not passed in.
> - A new Key Protect instance.
> 
> The output of the example will be the CRN of the Key Protect instance and the parsed fields of the CRN using the CRN parser submodule.

Reviewer: ```suggestion
This example provisions the following resources:

- A new resource group, if one is not passed in.
- A new Key Protect instance.

The outputs are the CRN of the Key Protect instance and the fields of the CRN that are parsed by using the CRN parser utility.
```

Using `utility` instead of submodule to stay consistent with the main readme file.

But it looks like this won't need the KP instance, based on Conall's comment. So readme could have something instead like, 

> This example takes a CRN as the input. The outputs are the fields of the CRN that are parsed by using the CRN parser utility.

Reviewer: @MatthewLemmond You should use `segments` instead of fields for the parts of the CRN to stay internally consistent and consistent with the cloud docs.

# CRN parser example

> This example takes a CRN as the input. The outputs are the fields of the CRN that are parsed by using the CRN parser utility.

Reviewer: ```suggestion
This example takes a CRN as the input. The outputs are the segments of the CRN that are parsed by using the CRN parser utility.
```

just want to keep this consistent with modules/crn-parser/README.md
